### PR TITLE
openjdk: adding new package openjdk

### PR DESCRIPTION
**Pull request by snehring**
closes #2622

I got annoyed trying to install jdk about 20 min ago.

Anyway, it seems it's much easier to get a binary distribution of openjdk with version 11. 

This is basically entirely copied from the jdk package to hopefully provide similar functionality. There are binaries for windows and macos as well, but I'm not sure how to denote different downloads for different os versions.

### CODE REVIEW

**Comment by adamjstewart** [comment]
How do `jdk` and `openjdk` differ? Should `openjdk` be the default `java` provider?

There are a few places in the package where you need to replace `jdk` with `openjdk`.

**Comment by snehring** [comment]
> 
> 
> How do `jdk` and `openjdk` differ? Should `openjdk` be the default `java` provider?
I think there's still a place for jdk (the oracle closed source java implementation), but openjdk (an entirely opensource jdk implementation) currently only has version 11.0.1 out so It's not a great replacement for earlier versions of java. It looks like they're aiming to have at least an updated version of jdk8 available similarly at some point. 

I need to revisit that icedtea package at some point and it'll be good to potentially use this to compile it.

> 
> There are a few places in the package where you need to replace `jdk` with `openjdk`.
I'll get that fixed up tomorrow. Dangers of copy paste.

**Comment by alalazo** [comment]
Relates #2622

**Comment by snehring** [comment]
I suppose I should also clarify that the biggest difference between this and previous attempts at packaging openjdk is that this one is very lazy and is using the now easily available binary distribution provided at https://jdk.java.net/11/. It could be used to compile other open source java implementations (like icedtea) if desired.

**Comment by adamjstewart** [comment]
I'm fine with a binary package. That's what we have for `jdk`, and it seems to be working fine for most users. If `openjdk` becomes more stable with more releases, I would love to see it become the default so we don't have to auto-accept EULA's for users anymore.

It looks like all of the Python 2 unit tests are failing over some ASCII error. Can you make sure you didn't accidentally include any Unicode characters?

**Comment by hartzell** [comment]
There's also [Corretto](https://aws.amazon.com/corretto/), Amazon's repackaging of OpenJDK w/ long term support commitments and etc....  It's been making the new lately, e.g. [on Hacker News](https://news.ycombinator.com/item?id=18449506).

Seems like its part of an ongoing "dump Oracle" push at Amazon (they also just migrated their largest data warehouse off of Oracle into an Aurora-postgresql database, with great fanfare).

Not sure if that's an easier place to get an JDK or not....

**Comment by scheibelp** [comment]
Going to check if closing and reopening will resolve the test issues.

File: var/spack/repos/builtin/packages/openjdk/package.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
+# Copyright 2013-2018 Lawrence Livermore National Security, LLC and other
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/openjdk/package.py]: @snehring could you update 2018 to 2019? Otherwise this all looks good

**Comment by scheibelp** [comment]
Thanks!

**Pull request merged**